client/src/js/app/store/store.js

```diff
@@ -208,6 +207,9 @@ const store = new Vuex.Store({
         })
       })
     },
+    updateLoadingState({ commit }, payload) {
```

Comment:
I don't think we need an action to set the loading state when we have a mutation that does the job. It's not an asynchronous task so components can just set the loading flag through the mutation.